extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java

```diff
@@ -167,6 +178,9 @@ public long count() {
         if (this.projections != null) {
             find.projection(projections);
         }
+        if (this.collation != null) {
```

Comment:
Is there a good reason the reactive/non-reactive queries have code that is different here? reactive has 2 `buildOptions` methods while here it's done in `list`. Is is normal or a sign that one of the two has a refactoring missing?

Reply:
Only the reactive API have the ability to pass an option to a Query.
The imperative must mutate the query before iteratinge the cursor.